AP-protocol.md

```diff
@@ -130,28 +127,41 @@ struct FloodRequest {
 ### **Neighbor Response**
 
 When a neighbor node receives the flood request, it processes it based on the following rules:
+- If the flood ID has already been received:
+	- The drone adds itself to the `path_trace`.
+	- The drone creates a `FloodResponse` and sends it back.
 
-- If the flood request was not received earlier, the node forwards the updated packet to its neighbors (except the one it received the flood request from) decreasing the TTL by 1, otherwise set the TTL to 0.
-- If the TTL of the message is 0, build a `FloodResponse` and send it along the same path back to the initiator.
+- If the flood ID has not yet been received:
+	- The drone adds itself to the `path_trace`.
+	- **If it has neighbors** (excluding the one from which it received the `FloodRequest`):
+		- The drone forwards the packet to its neighbors (except the one from which it received the `FloodRequest`).
+	- **If it has no neighbors**, then:
+		- The drone creates a `FloodResponse` and sends it to the node from which it received the `FloodRequest`.
 
 ```rust
 struct FloodResponse {
 	flood_id: u64,
-	source_routing_header: SourceRoutingHeader,
 	path_trace: Vec<(NodeId, NodeType)>
 }
 ```
 
+#### Notes:
+- For the discovery protocol, `Packet`s of type `FloodRequest` and `FloodResponse` will be sent.
+- The `routing_header` of `Packet`s of type `FloodRequest` will be ignored (as the Packet is sent to all neighbors except the one from which it was received).
+- The `routing_header` of `Packet`s of type `FloodResponse`, on the other hand, determines the packet's path.
+
 ### **Recording Topology Information**
 
-For every flood response or acknowledgment the initiator receives, it updates its understanding of the graph:
+For every flood response the initiator receives, it updates its understanding of the graph:
 
 - If the node receives a flood response with a **path trace**, it records the paths between nodes. The initiator learns not only the immediate neighbors but also the connections between nodes further out.
 - Over time, as the query continues to flood, the initiator accumulates more information and can eventually reconstruct the entire graph's topology.
 
 ### **Termination Condition**
 
 The flood can terminate when:
+- A node receives a `FloodRequest` with a flood_id that has already been received.
```

Comment:
```suggestion
- A node receives a `FloodRequest` with a flood_id that has already been received.
```
If a drone builds only one `FloodResponse` from receiving multiple FloodRequest from different edges this algorithm does not allow to discover all the edges. Instead, if you create a `FloodResponse` every time that the `flood_id` has already been received but it has different incoming edge (not seen yet) we should be able to discover all of the available edges.

Reply:
yes, a `FloodResponse` is created each time an already received `flood_id` is received.

Reply:
If we say each time, without checking the incoming edge, then we might be sending redundant information to the flooding initiator. Anyway, i'd specify it, since I don't think this really clear this way.